applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h

```diff
@@ -560,6 +560,11 @@ class TrilinosBlockBuilderAndSolver
         }
 
         temp_dofs_array.Unique();
+
+        if (mSynchronizeDofSet) {
```

Comment:
why do we need this here? in principle dofs should be already in sync when we are at this point (dofs should be always at sync after update in fact)

Reply:
i had a short offline conversation with @jcotela, but i am still very much convinced that dofs should be considered in sync when you are outside of the builder and solver. If they are not, they should be made so before entering there, otherwise you are fixing something in the wrong point

Reply:
Hi @RiccardoRossi I checked the code locally after the call and line 564 above (the entry point for the new functionality in the builder and solver) belongs to SetUpDofSet, not to SystemSolveWithPhysics as it would seem from the changelog. The extra cost is incurred once per system set-up not per system solve, which I agree with you that would be wrong.

The changes in mpi_communicator are needed because there is no method that can add dofs to the dofset of a different partition (the dofs are known to exist on that partition too, no actual transfer of dofs is happening).

Personally, I would prefer to leave the extra synchronization step as off by default, since using the standard partitioning strategy in Kratos should be enough to ensure that this situation never happens. Fixing the dofset here requires an extra communication step followed by an additional sort+unique of the dofs, which would happen even if no dofs were added.